Sasquatch/Sasquatch/ViewControllers/MSSignInViewController.swift

```diff
@@ -65,33 +48,35 @@ class MSSignInViewController: UIViewController, WKNavigationDelegate, MSAnalytic
 
   func login() {
     NSLog("Started login process")
-    if let signInUrl = URL(string: self.baseUrl + self.authorizeEndpoint + "?" + redirectParam + clientIdParam + "&response_type=token" + scopeParam) {
+    if let signInUrl = URL(string: kMSABaseUrl + kMSAAuthorizeEndpoint + "?" + kMSARedirectParam + kMSAClientIdParam + "&response_type=token" + kMSAScopeParam) {
       self.webView.load(URLRequest(url: signInUrl))
     }
   }
 
   func signOut() {
     NSLog("Started sign out process")
-    if let url = URL(string: self.baseUrl + self.signOutEndpoint + "?" + redirectParam + clientIdParam) {
+    if let url = URL(string: kMSABaseUrl + kMSASignOutEndpoint + "?" + kMSARedirectParam.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)! + kMSAClientIdParam) {
       self.webView.load(URLRequest(url: url))
     }
   }
 
   // The sign in flow.
   func signIn(url: URL) {
-    if url.absoluteString.contains((self.baseUrl + self.redirectEndpoint)) {
-      if let newUrl = URL(string: self.baseUrl + self.redirectEndpoint + "?" + url.fragment!) {
+    if url.absoluteString.contains((kMSABaseUrl + kMSARedirectEndpoint)) {
+      if let newUrl = URL(string: kMSABaseUrl + kMSARedirectEndpoint + "?" + url.fragment!) {
```

Comment:
<img width="852" alt="Screen Shot 2020-03-26 at 2 43 58 PM" src="https://user-images.githubusercontent.com/623672/77699817-5584ac80-6f70-11ea-9ed1-f0782ff0990c.png">

It crashes on simulator iphone XR / XCode 10.1

Reply:
Strange, my change just moves constants..

Reply:
Ok found what was the case